assets/dev/js/admin/pages/main-page.js

```diff
@@ -0,0 +1,340 @@
+
+	return (
+		<SnackbarList
+			className="edit-site-notices"
+			notices={ notices }
+			onRemove={ removeNotice }
+		/>
+	);
+};
+export const MainPage = () => {
+	const [ hasLoaded, setHasLoaded ] = useState( false );
+	const [ settingsData, setSettingsData ] = useState( {} );
+
+	const settingsPrefix = 'hello_elementor_settings';
+
+	/**
+	 * Update settings data.
+	 *
+	 * @param {string} settingsName
+	 * @param {string} settingsValue
+	 */
+	const updateSettings = ( settingsName, settingsValue ) => {
+		setSettingsData( {
+			...settingsData,
+			[ settingsName ]: settingsValue,
+		} );
+	};
+
+	/**
+	 * Save settings to server.
+	 */
+	const saveSettings = () => {
+		const settings = new api.models.Settings( {
+			[ settingsPrefix + '_description_meta_tag' ]: settingsData._description_meta_tag ? 'true' : '',
+			[ settingsPrefix + '_skip_link' ]: settingsData._skip_link ? 'true' : '',
+			[ settingsPrefix + '_page_title' ]: settingsData._page_title ? 'true' : '',
+			[ settingsPrefix + '_generator' ]: settingsData._generator ? 'true' : '',
+			[ settingsPrefix + '_shortlink' ]: settingsData._shortlink ? 'true' : '',
+			[ settingsPrefix + '_wlw' ]: settingsData._wlw ? 'true' : '',
+			[ settingsPrefix + '_rsd' ]: settingsData._rsd ? 'true' : '',
+			[ settingsPrefix + '_oembed' ]: settingsData._oembed ? 'true' : '',
+			[ settingsPrefix + '_wp_sitemap' ]: settingsData._wp_sitemap ? 'true' : '',
+			[ settingsPrefix + '_post_prev_next' ]: settingsData._post_prev_next ? 'true' : '',
+			[ settingsPrefix + '_site_rss' ]: settingsData._site_rss ? 'true' : '',
+			[ settingsPrefix + '_comments_rss' ]: settingsData._comments_rss ? 'true' : '',
+			[ settingsPrefix + '_post_comments_rss' ]: settingsData._post_comments_rss ? 'true' : '',
+			[ settingsPrefix + '_emoji' ]: settingsData._emoji ? 'true' : '',
+			[ settingsPrefix + '_jquery_migrate' ]: settingsData._jquery_migrate ? 'true' : '',
+			[ settingsPrefix + '_oembed_script' ]: settingsData._oembed_script ? 'true' : '',
+			[ settingsPrefix + '_classic_theme_styles' ]: settingsData._classic_theme_styles ? 'true' : '',
+			[ settingsPrefix + '_gutenberg' ]: settingsData._gutenberg ? 'true' : '',
+			[ settingsPrefix + '_hello_style' ]: settingsData._hello_style ? 'true' : '',
+			[ settingsPrefix + '_hello_theme' ]: settingsData._hello_theme ? 'true' : '',
+			
+		} );
+
+		settings.save();
+
+		dispatch( 'core/notices' ).createNotice(
+			'success',
+			__( 'Settings Saved', 'hello-elementor' ),
+			{
+				type: 'snackbar',
+				isDismissible: true,
+			},
+		);
+	};
+
+	useEffect( () => {
+		const fetchSettings = async () => {
+			try {
+				await api.loadPromise;
+				const settings = new api.models.Settings();
+				const response = await settings.fetch();
+				setSettingsData( {
+					_description_meta_tag: response[ settingsPrefix + '_description_meta_tag' ],
+					_skip_link: response[ settingsPrefix + '_skip_link' ],
+					_page_title: response[ settingsPrefix + '_page_title' ],
```

Comment:
please move setting names to a single const object and reference from there
```JS
const SETTINGS = {
 DESCRIPTION_META_TAG: 'description_meta_tag',
};
```

Reply:
Done.